howso.amlg

```diff
@@ -1,4073 +1,786 @@
+		(call !Return)
+	)
 
-			;if there are no encoded features, run the raw ExportCasesAndSessions call
-			(if (= (false) has_encoded_features has_rounded_features has_datetime_features)
-				(conclude (call !export_cases_and_sessions))
-			)
-
-			(store
-				(concat migration_filepath trainee ".exp.json")
-				(map
-					(lambda (let
-						(assoc
-							;assoc of feature -> feature value
-							entity_code_map	(get_all_labels (retrieve_entity_root (list trainee (current_index 2)) (true)) )
-						)
+	;get metadata for model
+	#get_metadata
+	(call !Return (assoc payload (retrieve_from_entity "!metaData") ))
 
-						;"case" entity has a label named ".session", decode values as needed
-						(if (contains_index entity_code_map ".session")
-							(zip
-								(indices entity_code_map)
-								;decode the values accordingly
-								(if has_encoded_features
-									(call_entity trainee "ConvertToOutput" (assoc
-										features (indices entity_code_map)
-										feature_values (values entity_code_map)
-										has_rounded_features has_rounded_features
-										has_datetime_features has_datetime_features
-									))
-
-									has_rounded_features
-									(call_entity trainee "RoundContinuousFeatures" (assoc
-										features (indices entity_code_map)
-										feature_values (values entity_code_map)
-									))
-								)
-							)
+	;set the random seed on a trainee
+	;parameters:
+	; seed : optional, the value of the random seed to set on the trainee, defaults to a system-provided 64-bit random number
+	#set_random_seed
+	(declare
+		(assoc seed (system "rand" 16) )
 
-							;else non-case entity, output as-is
-							entity_code_map
-						)
-					))
-					(zip (contained_entities trainee))
-				)
-			)
-		)
+		(set_entity_rand_seed trainee seed)
+		(accum_to_entities (assoc !revision 1))
 
+		(call !Return)
+	)
 
-	;Method to re-create the featureCustomDerivedMethods label from attributes for a trainee, used when upgrading a trainee
-	#!recreate_feature_custom_derived_methods
-		(let
-			(assoc
-				custom_derived_features_map
-					(filter
-						(lambda (or
-							(!= (null) (get (current_value) "auto_derive_on_train"))
-							(!= (null) (get (current_value) "derived_feature_code"))
-						))
-						(retrieve_from_entity trainee "featureAttributes")
-					)
-			)
-
-			;only if there are derived features
-			(if (size custom_derived_features_map)
-				(let
-					(assoc
-						;filter features, leaving only those with custom derivations
-						custom_derived_features_map
-							(filter
-								(lambda (or
-									(= "custom" (get (current_value) (list "auto_derive_on_train" "derive_type")))
-									(!= (null) (get (current_value) "derived_feature_code"))
-								))
-								custom_derived_features_map
-							)
-					)
+	;set the influence weight threshold for outputting only the K neighbors whose inflence weight is <= to this threshold
+	;default value is 0.99
+	#set_influence_weight_threshold
```

Comment:
should this be changed to a proper `(declare (assoc` pattern?

I suggest this so that we can actually set this 0.99 default value. Otherwise calling it without the value will assign a `(null)`. Also should doc the parameter here I guess